client/app/pages/dashboards/ShareDashboardDialog.jsx

```diff
@@ -0,0 +1,122 @@
+import { replace } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import Switch from 'antd/lib/switch';
+import Modal from 'antd/lib/modal';
+import Form from 'antd/lib/form';
+import Tooltip from 'antd/lib/tooltip';
+import { $http, toastr } from '@/services/ng';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import InputWithCopy from '@/components/InputWithCopy';
+
+const API_SHARE_URL = 'api/dashboards/{id}/share';
+const HELP_URL = 'https://redash.io/help/user-guide/dashboards/sharing-dashboards?source=dialog';
+
+class ShareDashboardDialog extends React.Component {
+  static propTypes = {
+    dashboard: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    dialog: DialogPropType.isRequired,
+  };
+
+  formItemProps = {
+    labelCol: { span: 8 },
+    wrapperCol: { span: 16 },
+    style: { marginBottom: 7 },
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      saving: false,
+    };
+    this.apiUrl = replace(API_SHARE_URL, '{id}', props.dashboard.id);
+  }
+
+  static get headerContent() {
+    return (
+      <React.Fragment>
+        Share Dashboard
+        <div className="modal-header-desc">
+          Allow public access to this dashboard with a secret address.{' '}
+          <Tooltip title="Guide: Sharing and Embedding Dashboards">
+            { /* eslint-disable-next-line react/jsx-no-target-blank */}
+            <a href={HELP_URL} target="_blank" rel="noopener">Learn more</a>
```

Comment:
One last question I forgot :sweat_smile:: why not `rel="noopener noreferrer"` to avoid `eslint-disable`? ([ref](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-target-blank.md))

Reply:
Heh heh :) Cause I think it might be valuable for us to track the origin of visits to help pages.
That's what the referrer is for so I don't want to mess up the stats.

Reply:
@kravets-levko @gabrieldutra legit?

Reply:
Yes, the vulnerability seems to be with the lack of `noopener`. Anyway, I think we can trust redash.io website :slightly_smiling_face:

Reply:
Exactly :)